File: src/main/java/no/nav/tpregisteret/domain/TpOrdning.java
package no.nav.tpregisteret.domain;

public class TpOrdning {
    private String tssId;
    private String tpId;
    private String orgNr;
    private String navn;

    public TpOrdning() {
    }

    public TpOrdning(String tssId, String tpId, String orgNr, String navn) {
        this.tssId = tssId;
        this.tpId = tpId;
        this.orgNr = orgNr;
        this.navn = navn;
    }

    public String getTssId() {
        return tssId;
    }

    public void setTssId(String tssId) {
        this.tssId = tssId;
    }

    public String getTpId() {
        return tpId;
    }

    public void setTpId(String tpId) {
        this.tpId = tpId;
    }

    public String getOrgNr() {
        return orgNr;
    }

    public void setOrgNr(String orgNr) {
        this.orgNr = orgNr;
    }

    public String getNavn() {
        return navn;
    }

    public void setNavn(String navn) {
        this.navn = navn;
    }
}

File: src/main/kotlin/no/nav/tpregisteret/domain/TpOrdning.kt
package no.nav.tpregisteret.domain

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "T_TSS_TP")
class TpOrdning(
    @Suppress("unused") @Column(name = "TSS_ID")
    @Id
    val tssId: String? = null,

    @Column(name = "TP_ID")
    val tpId: String,

    @Column(name = "ORGNR")
    val orgNr: String,

    @Column(name = "NAVN")
    val navn: String
) {
    override fun equals(other: Any?) = other is TpOrdning && other.tpId == tpId
    override fun hashCode() = tpId.hashCode()
}